add_node.rb

```diff
@@ -10,8 +11,37 @@
 fileData = File.read("data.json")
 obj = JSON.parse(fileData)
 
+# Initialise time periods.Dates have the format of yyyy-mm-dd
+this_year = Date.today.strftime("%Y")
+MID_SPRING_PERIOD = (this_year+"-02-15 to "+this_year+"-04-30").split(" to ")
+END_SPRING_PERIOD = (this_year+"-05-01 to "+this_year+"-09-30").split(" to ")
+MID_AUTUMN_PERIOD = (this_year+"-10-01 to "+this_year+"-11-30").split(" to ")
+END_AUTUMN_PERIOD = (this_year+"-12-01 to "+(this_year.to_i+1).to_s+"-02-14").split(" to ")
```

Comment:
So, in these 4 lines, the `split` is unnecessary. Just change this to a list and also, please use the `#{this_year}-MM-dd` notation as that is more intuitive to read than string addition.